en/codegen/vscode-plugin.md

```diff
@@ -0,0 +1,155 @@
+---
+title: Indigo.Design Code Generator for VS Code
+_description: Quick overview on how the extension works and how to use it.
+_keywords: Design System, Code Generation, Ignite UI for Angular, Component, UI Library, Widgets, Routing, VS Code, Extension
+---
+
+## Indigo.Design Code Generator for VS Code
+
+This topic describes the Visual Studio Code extension that we have built to make code generation as easy as possible for developers.  With this extension developers can generate actual, runnable code for Angular projects directly from within Visual Studio Code.  Designers can take their Sketch files and publish them to the cloud with [Indigo.Design](https://next.indigo.design) and then provide developers with a share link.  Developers can take this share link, provide it directly to this extension and generate code off the design saving them tons of time that would otherwise be spent manually building modules and writing HTML, CSS and Typescript for components.
+
+### Requirements
+
+This extension has the following dependencies.  Please install them before using.
+* [Node.js](https://nodejs.org/)
+* [npm](https://www.npmjs.com/)
+* [Angular CLI](https://cli.angular.io/) (installed globally)
+
+This extension also requires an Angular project to currently be opened in the workspace and at the root of the workspace.  This means that the `angular.json` or `.angular-cli.json` (depending on what version of Angular you are using) must be at the root of the workspace.
+
+The code generator produces [SCSS](https://sass-lang.com/) so it is recommended that your Angular project is setup to use it.  If you are creating a new Angular project you can set this up with the following:
+
+```
+ng new my-app --style=scss
+```
+
+If you are working with an existing Angular project then you can update the default style extension attribute in the Angular json file.
+
+**angular.json**
+```json
+"@schematics/angular:component": {
+    "styleext": "scss"
+}
+```
+**.angular-cli.json**
+```json
+"defaults": {
+    "styleExt": "scss",
+}
+```
+
+This will tell the Angular CLI to generate SCSS instead of CSS when generating components which this extension will use.
+
+If you are working with an existing Angular project and it is already using the [Ignite UI for Angular](https://www.npmjs.com/package/igniteui-angular) package then you will need to update to at least version 6.1.0.  This extension may not update your project correctly if you have an older version.
+
+## How it works
+
+It all starts with a sketch.  Once a designer has completed their sketch they can upload it to the [Indigo.Design](https://next.indigo.design) website.  From there the designer can publish their work and create usability studies or open it up for commenting.  When the work is published Indigo.Design will provide a link to allow others to view your work.  That link can also be used by this extension.
+
+![](https://dl.infragistics.com/tools/extensions/codegen/step1.gif)
```

Comment:
Is it ok to use an image from dl.infragistics.com?  The other images in the help are local.  If this works, it's fine.

Reply:
Microsoft requires that images in extension readmes come from https urls so we had setup a location for image hosting a while ago.  I can change this to include the images here.  That's just the way I had it setup and it works so didn't bother changing it.

I think we should be consistent with the rest of the docs though and include the images in github.